extensions/amp-story/1.0/amp-story-reaction.js

```diff
@@ -520,47 +504,37 @@ export class AmpStoryReaction extends AMP.BaseElement {
    * @private
    */
   handleSuccessfulDataRetrieval_(response) {
-    if (!(response && 'data' in response)) {
-      dev().error(
-        TAG,
-        `Invalid reaction response, expected { data: ReactionResponseType, ...} but received ${response}`
-      );
-      return;
-    }
-
-    this.responseData_ = response.data;
-
-    this.hasUserSelection_ = this.responseData_.hasUserResponded;
-    if (this.hasUserSelection_) {
-      this.updateReactionOnDataRetrieval_(response.data);
-    }
+    devAssert(
+      response && 'options' in response,
+      `Invalid reaction response, expected { data: ReactionResponseType, ...} but received ${response}`
+    );
+    const numOptions = this.rootEl_.querySelectorAll(
+      '.i-amphtml-story-reaction-option'
+    ).length;
+    this.updateReactionOnDataRetrieval_(
+      response['options'].splice(0, numOptions)
+    );
   }
 
   /**
    * Updates the quiz to reflect the state of the remote data.
-   * @param {!ReactionResponseType} data
+   * @param {!Array<ReactionOptionType>} data
    * @private
    */
   updateReactionOnDataRetrieval_(data) {
-    let selectedOptionKey;
-    /** @type {!Array} */ (data['responses']).forEach((response) => {
-      if (response.selectedByUser) {
-        selectedOptionKey = response.reactionValue;
-      }
-    });
-
-    if (selectedOptionKey === undefined) {
-      dev().error(TAG, `The user-selected reaction could not be found`);
```

Comment:
I think we shouldn't ever see this error, since this method assumed hasUserSelection=true but the backend doesn't have that field any more. The error essentially was catching the hasUserSelection=true and no option to be selected, which won't happen with the new API design.